OpenSim/Common/Exception.h

```diff
@@ -67,14 +91,29 @@ class OSIMCOMMON_API Exception  : public std::exception {
     std::string _file;
     /** Line number at which the error occurred. */
     int _line;
+    /** Delimiter/separator used to separate different parts of the message.  */
+    static const std::string _msgSeparator;
 
 //=============================================================================
 // METHODS
 //=============================================================================
 public:
     // CONSTRUCTORS
-    Exception(const std::string &aMsg="",const std::string &aFile="",int aLine=-1);
+    Exception(const std::string &aMsg="",
+              const std::string &aFile="",
+              int aLine=-1);
+
+    /** Call this constructor from derived classes to add file, line and 
+    function information to the error message                                 */
+    Exception(const std::string& file,
```

Comment:
I also think that the constructor should take a message and that is appended to to what the *base* message you are generating with formatted file, line, func info instead of having to call `addMessage`. Then derived Exceptions just supply what they they need to add at construction.